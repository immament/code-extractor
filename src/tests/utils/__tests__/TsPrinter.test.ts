import {NodeStub} from '@tests/stubs/NodeStub';

import ts from 'typescript';
import {createSourceFile} from '../createSourceFile';
import {createReference} from '../stubCreators';
import {TsPrinter} from '../TsPrinter';

describe('TsPrinter with default options', () => {
  let printer: TsPrinter;

  beforeEach(() => {
    printer = new TsPrinter();
  });

  test('should printNode without childs result match snampshot', () => {
    const result = printer.printNode(
      new NodeStub({kind: ts.SyntaxKind.ClassDeclaration}).asNode()
    );

    expect(result).toMatchSnapshot();
  });

  test('should printNodeWithoutChilds result match snampshot', () => {
    const result = printer.printNodeWithoutChilds(
      new NodeStub({
        kind: ts.SyntaxKind.ClassDeclaration,
        childs: [new NodeStub({kind: ts.SyntaxKind.Identifier})],
      }).asNode()
    );

    expect(result).toMatchSnapshot();
  });

  test('should print node with childs', () => {
    const sourceFile = createSourceFile(
      '/index.ts',
      'class MyClass { myMethod() {}}'
    );

    const result = printer.printNode(sourceFile);

    expect(result).toMatchSnapshot();
  });

  test('should print references', () => {
    const references = [
      createReference(),
      createReference(),
      createReference(),
    ];
    expect(printer.printReferences(references)).toMatchSnapshot();
  });
});

describe('TsPrinter with changed options', () => {
  let printer: TsPrinter;

  beforeEach(() => {
    printer = new TsPrinter({
      joinLineCharacter: '-',
      textFragmentLength: 20,
      useColors: false,
    });
  });

  test('should printNode without childs result match snampshot', () => {
    const result = printer.printNode(
      new NodeStub({kind: ts.SyntaxKind.ClassDeclaration}).asNode()
    );

    expect(result).toMatchSnapshot();
  });

  test('should printNodeWithoutChilds result match snampshot', () => {
    const result = printer.printNodeWithoutChilds(
      new NodeStub({
        kind: ts.SyntaxKind.ClassDeclaration,
        childs: [new NodeStub({kind: ts.SyntaxKind.Identifier})],
      }).asNode()
    );

    expect(result).toMatchSnapshot();
  });

  test('should print node with childs', () => {
    const sourceFile = createSourceFile(
      '/index.ts',
      'class MyClass { myMethod() {}}'
    );

    const result = printer.printNode(sourceFile);

    expect(result).toMatchSnapshot();
  });

  test('should print references', () => {
    const references = [
      createReference(),
      createReference(),
      createReference(),
    ];
    expect(printer.printReferences(references)).toMatchSnapshot();
  });
});
