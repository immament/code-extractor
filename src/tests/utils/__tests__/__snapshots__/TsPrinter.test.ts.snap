// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TsPrinter with changed options should print node with childs 1`] = `"SourceFile class MyClass { myMe- ClassDeclaration class MyClass { myMe-  Identifier MyClass-  MethodDeclaration myMethod() {}-   Identifier myMethod-   Block {}- EndOfFileToken "`;

exports[`TsPrinter with changed options should print references 1`] = `"From node: Identifier text-78-FROM:-  ClassDeclaration text-252-   Identifier text-78-   MethodDeclaration text-165-TO:-  FirstStatement text-232-From node: Identifier text-78-FROM:-  ClassDeclaration text-252-   Identifier text-78-   MethodDeclaration text-165-TO:-  FirstStatement text-232-From node: Identifier text-78-FROM:-  ClassDeclaration text-252-   Identifier text-78-   MethodDeclaration text-165-TO:-  FirstStatement text-232"`;

exports[`TsPrinter with changed options should printNode without childs result match snampshot 1`] = `"ClassDeclaration text-252"`;

exports[`TsPrinter with changed options should printNodeWithoutChilds result match snampshot 1`] = `"ClassDeclaration text-252"`;

exports[`TsPrinter with default options should print node with childs 1`] = `
"SourceFile
 ClassDeclaration
  Identifier
  MethodDeclaration
   Identifier
   Block
 EndOfFileToken"
`;

exports[`TsPrinter with default options should print references 1`] = `
"[36mFrom node: [39mIdentifier
[36mFROM:[39m
  ClassDeclaration
   Identifier
   MethodDeclaration
[36mTO:[39m
  FirstStatement
[36mFrom node: [39mIdentifier
[36mFROM:[39m
  ClassDeclaration
   Identifier
   MethodDeclaration
[36mTO:[39m
  FirstStatement
[36mFrom node: [39mIdentifier
[36mFROM:[39m
  ClassDeclaration
   Identifier
   MethodDeclaration
[36mTO:[39m
  FirstStatement"
`;

exports[`TsPrinter with default options should printNode without childs result match snampshot 1`] = `"ClassDeclaration"`;

exports[`TsPrinter with default options should printNodeWithoutChilds result match snampshot 1`] = `"ClassDeclaration"`;
